grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java

```diff
@@ -68,19 +68,10 @@
     this.stopwatch = checkNotNull(stopwatch, "stopwatch");
     this.backoffPolicyProvider = checkNotNull(backoffPolicyProvider, "backoffPolicyProvider");
     this.subchannelPool = checkNotNull(subchannelPool, "subchannelPool");
-    this.subchannelPool.init(helper, this);
```

Comment:
Because we removed `init`, we don't need the `subchannelPool` field in this class. We can make `subchannelPool` a final field of `GrpclbState`.

Reply:
good catch, i didn't do it because it makes very hard to convert existing tests because grpclb tests mocks the subchannelpool and passed it through `GrpclbLoadBalancer`. alternatively, I could create a provider or factory class which i thought little too much, wdyt?

Reply:
The current test approach seems terribly wrong:
```java
      doAnswer(new Answer<Subchannel>() {
        @Override
        public Subchannel answer(InvocationOnMock invocation) throws Throwable {
          Subchannel subchannel = mock(Subchannel.class);
          EquivalentAddressGroup eag = (EquivalentAddressGroup) invocation.getArguments()[0];
          Attributes attrs = (Attributes) invocation.getArguments()[1];
          when(subchannel.getAllAddresses()).thenReturn(Arrays.asList(eag));
          when(subchannel.getAttributes()).thenReturn(attrs);
          mockSubchannels.add(subchannel);
          pooledSubchannelTracker.add(subchannel);
          return subchannel;
        }
      }).when(subchannelPool).takeOrCreateSubchannel(
          any(EquivalentAddressGroup.class), any(Attributes.class));
```

I think a more correct way to test is not mocking subchannelPool, but just mocking `helper.createSubchannel()` and keeping track of the EAGs and subchannels. But then a lot of current tests need be rewritten. 

I'm OK with using a factory in the short term if it's too hard to fix tests, but it worth rewriting some of the tests in the future.

Reply:
i do agree that current test is not really testing anything. at best, it is testing it's own implementation via mock. fixing this (by making fake classes or using real classes instead of mock) is essentially rewriting the entire test.
i'll use the factory approach, i am not sure how much time we want to invest to fix the test though given the state of the grpclb.

Reply:
actually i like the factory better, because the pool is ignored/unused when it is using pick_first. using factory prevents it to create unnecessary pool. sounds like win-win.